kernel/src/utilities/capability_ptr.rs

```diff
@@ -66,42 +60,45 @@ impl From<usize> for CapabilityPtr {
     /// Provenance note: may have null provenance.
     #[inline]
     fn from(from: usize) -> Self {
-        Self {
-            ptr: from as *const (),
-        }
+        Self { ptr: from.into() }
+    }
+}
+
+impl From<MachineRegister> for CapabilityPtr {
```

Comment:
This is a narrowing conversion and therefore should be a method rather than a `From` impl. IMO this should be a method on `MachineRegister` but it could be on `CapabilityPtr` as well.

Also, shouldn't there be a `From<CapabilityPtr> for MachineRegister` impl?

Reply:
Moved to a `as_capability_ptr` method in #4332 .